flycheck.el

```diff
@@ -1803,14 +1802,15 @@ CHECKER is a checker symbol.
 
 Pop up a help buffer with the documentation of CHECKER."
   (interactive
-   (let* ((checker (flycheck-checker-at-point))
+   (let* ((checker-at-pt (flycheck-checker-at-point))
           (enable-recursive-minibuffers t)
-          (prompt (if (symbolp checker)
-                      (format "Describe syntax checker (default %s): " checker)
-                    "Describe syntax checker: "))
-          (reply (read-flycheck-checker prompt)))
-     (list (or reply checker))))
-  (if (or (null checker) (not (flycheck-valid-checker-p checker)))
+          (default (or checker-at-pt
+                       (ignore-errors (flycheck-get-checker-for-buffer))))
+          (prompt (if default
+                      (format "Describe syntax checker (default %s): " default)
+                    "Describe syntax checker: ")))
+     (list (read-flycheck-checker prompt default))))
+  (if (not (flycheck-valid-checker-p checker))
       (message "You didn't specify a Flycheck syntax checker.")
```

Comment:
While we are at it, shouldn't this be a `user-error`?  I think that's the better choice here, isn't it?